src/net/tcp/listener.rs

```diff
@@ -40,15 +40,43 @@ impl TcpListener {
     ///
     /// Binding with a port number of 0 will request that the OS assigns a port
     /// to this listener.
-    ///
-    /// In the future, the port allocated can be queried via a (blocking) `local_addr`
-    /// method.
     pub fn bind(addr: SocketAddr) -> io::Result<Self> {
         let socket = Socket::bind(addr, libc::SOCK_STREAM)?;
         socket.listen(1024)?;
         Ok(TcpListener { inner: socket })
     }
 
+    /// Returns the local address that this listener is bound to.
+    ///
+    /// This can be useful, for example, when binding to port 0 to figure out
```

Comment:
I wouldn't exactly call this blocking. Certain system calls (like this one) are fast enough that they can be considered nonblocking. This one will simply drop to kernelspace to grab some data about the socket and return back.

It's probably even less expensive than another syscall we generally agree to be nonblocking, the `recv` syscall (in nonblocking mode), which does far more work in kernelspace than this one!

Reply:
Agreed.